### PR TITLE
PRIME-1981 Review use of getEnrolleeById in admin

### PR DESCRIPTION
**Pull request by johnnyl6**
- refactored to use individual `getAccessAgreementNote` method instead of `getEnrolleeById`
- added email field to `EnrolleeListViewModel` so no extra population of email is needed by calling `getEnrolleeById`
- removed unnecessary `getEnrolleeById` call in `enrollee-access-term-enrolment.component.ts`
- made separate viewModel to contain `enrolmentStatuses`, `selfDeclarationDocuments` and `identificationDocuments`
- removed related api calls inside `forkjoin` of `getEnrolleeById` in `adjudicationResource`
- triange component performance improvement
- disable sorting queryParams for single-row enrollee
- made queryParams not applicable to single-row enrolle
- other minor improvements

### CODE REVIEW

File: prime-dotnet-webapi/ViewModels/Enrollee/EnrolleeListViewModel.cs [hunk 1]
```diff
@@ -76,5 +76,7 @@ public string CurrentTOAStatus
         public bool PossiblePaperEnrolmentMatch { get; set; }
 
         public DateTime DateOfBirth { get; set; }
+
+        public string Email { get; set;}
```
[the remark this answers is not in the document]
Comment (johnnyl6) [reply prime-dotnet-webapi/ViewModels/Enrollee/EnrolleeListViewModel.cs:80]: > Where is the Emil being used from the `EnrolleeListViewModel` ?

In the hamburger menu in enrollee table, there's a `Send Email`. Previously each click would trigger a `getEnrolleeById` call only for the email address. Now we populate the email to the table directly and send the email by EventEmitter without calling `getEnrolleeById`.

**Review by AMiKaH — APPROVED** [review]
LGTM

File: prime-angular-frontend/src/app/modules/adjudication/pages/enrollee-access-term-enrolment/enrollee-access-term-enrolment.component.ts [hunk 2]
```diff
@@ -41,20 +39,10 @@ export class EnrolleeAccessTermEnrolmentComponent extends AbstractComponent impl
...
-            this.enrolleeSubmissionAdapterResponse(enrolleeSubmission),
-            this.enrolleeAdapterResponse(enrolment)
-        ]
-      )
-    ).subscribe(([enrolmentSubmission, enrolment]: [EnrolmentSubmission, Enrolment]) => {
-      this.enrolmentSubmission = enrolmentSubmission;
-      this.enrolment = enrolment;
-    });
+    this.busy = this.adjudicationResource.getSubmissionForAgreement(enrolleeId, accessTermId)
+      .subscribe((enrolmentSubmission) =>
```
Comment (hogmann) [inline prime-angular-frontend/src/app/modules/adjudication/pages/enrollee-access-term-enrolment/enrollee-access-term-enrolment.component.ts:43]: Can we make sure to have typing whenever we can
``` (enrolmentSubmission: EnrolmentSubmission) =>```

File: prime-angular-frontend/src/app/modules/adjudication/pages/enrolment/enrolment.component.ts [hunk 3]
```diff
@@ -38,19 +34,9 @@ export class EnrolmentComponent extends AbstractComponent implements OnInit {
...
-    this.busy = this.adjudicationResource.getEnrolleeById(enrolleeId)
-      .pipe(
-        map((enrollee: HttpEnrollee) => this.enrolleeAdapterResponse(enrollee))
-      )
-      .subscribe((enrollee: Enrolment) => this.enrollee = enrollee);
-  }
+  public ngOnInit() { }
 
-  private enrolleeAdapterResponse(enrollee: HttpEnrollee): Enrolment {
+  public enrolleeAdapterResponse(enrollee: HttpEnrollee): Enrolment {
```
Comment (hogmann) [inline prime-angular-frontend/src/app/modules/adjudication/pages/enrolment/enrolment.component.ts:39]: all public methods should be above ngOnInit

File: prime-angular-frontend/src/app/modules/adjudication/shared/components/adjudication-container/adjudication-container.component.ts [hunk 4]
```diff
@@ -331,6 +329,7 @@ export class AdjudicationContainerComponent implements OnInit {
       )
       .subscribe((enableEnrollee: HttpEnrollee) => {
```
Comment (hogmann) [inline prime-angular-frontend/src/app/modules/adjudication/shared/components/adjudication-container/adjudication-container.component.ts:330]: So all the places but two (toggle always manual / assign TOA) we call these two methods which both have get enrolleeById
image
Probably easiest thing to so is not call it in the adjudicationActionPipe and just use the update enrollee to handle updating the enrollee

File: prime-angular-frontend/src/app/modules/adjudication/pages/enrollee-access-term-enrolment/enrollee-access-term-enrolment.component.ts [hunk 5]
```diff
@@ -41,20 +39,10 @@ export class EnrolleeAccessTermEnrolmentComponent extends AbstractComponent impl
...
-        [
-            this.enrolleeSubmissionAdapterResponse(enrolleeSubmission),
-            this.enrolleeAdapterResponse(enrolment)
-        ]
-      )
-    ).subscribe(([enrolmentSubmission, enrolment]: [EnrolmentSubmission, Enrolment]) => {
-      this.enrolmentSubmission = enrolmentSubmission;
-      this.enrolment = enrolment;
-    });
+    this.busy = this.adjudicationResource.getSubmissionForAgreement(enrolleeId, accessTermId)
```
Comment (hogmann) [inline prime-angular-frontend/src/app/modules/adjudication/pages/enrollee-access-term-enrolment/enrollee-access-term-enrolment.component.ts:42]: We aren't consuming any of this data in the parent, so we can actually move this call into the child.

Comment (hogmann) [reply prime-angular-frontend/src/app/modules/adjudication/pages/enrollee-access-term-enrolment/enrollee-access-term-enrolment.component.ts:42]: This helps make the child more self contained and cleanups this fork join a little.

Comment (johnnyl6) [reply prime-angular-frontend/src/app/modules/adjudication/pages/enrollee-access-term-enrolment/enrollee-access-term-enrolment.component.ts:42]: looks like line 9 does have a reference: `enrolmentSubmission?.createdDate`
```HTML
<h6 class="text-green text-italic">Submitted {{ enrolmentSubmission?.createdDate | formatDate }}</h6>
```

**Review by Paahn — APPROVED** [review]
LGTM

File: prime-angular-frontend/src/app/modules/adjudication/pages/enrollee-access-term-enrolment/enrollee-access-term-enrolment.component.ts (hunk 5 above)
Comment (hogmann) [reply prime-angular-frontend/src/app/modules/adjudication/pages/enrollee-access-term-enrolment/enrollee-access-term-enrolment.component.ts:42]: closing this and moving to where it was supposed to be

File: prime-angular-frontend/src/app/modules/adjudication/pages/enrollee-overview/enrollee-overview.component.ts [hunk 6]
```diff
@@ -95,14 +97,16 @@ export class EnrolleeOverviewComponent extends AdjudicationContainerComponent im
             }))
           ),
         enrolleeNavigation: this.adjudicationResource.getAdjacentEnrolleeId(enrolleeId),
+        reviewStatus: this.adjudicationResource.getEnrolleeReviewStatus(enrolleeId)
```
Comment (hogmann) [inline prime-angular-frontend/src/app/modules/adjudication/pages/enrollee-overview/enrollee-overview.component.ts:100]: This is where i meant to make that comment, the reviewStatus is only being consumed by the "app-review-status-content" component so we can move this call to inside there.

File: prime-angular-frontend/src/app/modules/adjudication/shared/components/adjudication-container/adjudication-container.component.ts (hunk 4 above)
Comment (johnnyl6) [reply prime-angular-frontend/src/app/modules/adjudication/shared/components/adjudication-container/adjudication-container.component.ts]: Done

**Review by AMiKaH — APPROVED** [review]

**Review by hogmann — APPROVED** [review]
LGTM

**Pull request merged**